kernel/userspace.c

```diff
@@ -156,6 +156,50 @@ static size_t obj_size_get(enum k_objects otype)
 	return ret;
 }
 
+static size_t obj_align_get(enum k_objects otype)
+{
+	size_t ret;
+
+	switch (otype) {
+	case K_OBJ_THREAD:
+		if (IS_ENABLED(CONFIG_X86_64)) {
```

Comment:
I don't want arch-specific stuff like this in the core kernel.

Define some optional ARCH_THREAD_OBJECT_ALIGNMENT in arch.h as needed and use that.

Reply:
Updated to do what you suggested instead.